### DISCUSSION TITLE
Update in HA Docker

### DISCUSSION THREAD

**Discussion by Matssa56** [flobz/psa_car_controller#341] [Q&A]
Hi!

I'm using the HA addon based on the docker container. In order to update it, I need to uninstall it everytime and reconfigure everything. Is there a way to update the files directly in the container with a py command or something?

Thanks!

**Comment by flobz** [comment]
Why do you uninstall it ? You should have a upgrade button...

**Reply by Matssa56** [reply]
Hi,

No it doesn't show since the HA json config uses a specific docker hub image (version which is showed in HA).

Currently it's still on v2.6.0 and uses your docker hub image v2.6.0.

In order to update the HA addon you need to update the config.json to the new docker image in docker hub. FYI you can't use latest, you need to specify a version. I forked your addon repo but kinda hard to know which image version to use since there's a few v0 (g0, v0.0, etc...).

Since you're using python repo now, I can maybe go in the docker image and update the python repo there.

**Reply by flobz** [reply]
I intentionally didn't update the repo because the last version might be not stable.

**Reply by flobz** [reply]
And it's impossible to downgrade version if there is an issue.

**Reply by Matssa56** [reply]
Well the docker image used by the Home Assistant addon is 4months old and was regularly broken, I had to uninstall and reinstall the addon so that it fetches the latest GitHub files and all (at least I assume it worked like this). But doing so I had to reconfigure every time.

So if I understand correctly it is intended? Would it be possible to make a dev branch / addon / image so that I can test the latest GitHub additions (for example the change in installation through pip)?

**Reply by flobz** [reply]
Yes you can totally do it yourself
